src/libopenrave-core/stringutils.h

```diff
@@ -0,0 +1,95 @@
+// -*- coding: utf-8 -*-
+// Copyright (C) 2023 Tan Li Boon <liboon.tan@mujin.co.jp>
+//
+// This file is part of OpenRAVE.
+// OpenRAVE is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#ifndef RAVE_STRINGUTILS
+#define RAVE_STRINGUTILS
+
+#include <algorithm>
+#include <string>
+
+#if  __cplusplus >= 201703L
+#define OPENRAVE_NODISCARD [[nodiscard]]
+#else
+#define OPENRAVE_NODISCARD __attribute__((warn_unused_result))
+#endif
+
+namespace OpenRAVE {
+
+// Returns true if a string begins with a matching prefix.
+OPENRAVE_NODISCARD inline bool StringStartsWith(const std::string& input, const std::string& prefix, bool ignoreCase = true) {
```

Comment:
@undisputed-seraphim these are not very efficient implementations, so can you just put the header file close to the new code you write that will use them? or you can make them more efficient by reducing memory allocation (ie creating new std::strings)

Reply:
As discussed I will change it to `StringStartsWith(const char* inputStr, size_t inputSz, const char* prefixStr, size_t prefixSz, bool ignoreCase)`.